pkg/processor/trigger/nats/test/nats_test.go

```diff
@@ -0,0 +1,138 @@
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/dockerclient"
+	"github.com/nuclio/nuclio/pkg/errors"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/processor/test/suite"
+
+	"github.com/nats-io/go-nats"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	natsPort  = 4222
+	topicName = "my.topic"
+)
+
+type NatsTestSuite struct {
+	processorsuite.TestSuite
+	natsContainerId  string
+	natsSubscription *nats.Subscription
+	natsPort         int
+	topicName        string
+}
+
+func (suite *NatsTestSuite) SetupSuite() {
+	var err error
+
+	suite.TestSuite.SetupSuite()
+
+	suite.natsPort = natsPort
+	suite.topicName = topicName
+
+	// Start nats
+	suite.natsContainerId, err = suite.DockerClient.RunContainer("nats:latest",
```

Comment:
There might be a race condition here - perhaps NATS isn't ready before moving on. Please check if there's a simple way to check if NATS is operational before moving on (in RMQ, there's a sleep and a TODO ...)